mdx/plugins/codeTabsResolver.mjs

```diff
@@ -0,0 +1,152 @@
+ *
+ * @return {string | undefined} - The value related to the given key
+ */
+function getMetaParameter(meta, key) {
+  if (typeof meta !== 'string') {
+    return;
+  }
+
+  const matches = meta.match(new RegExp(`${key}="(?<parameter>[^"]*)"`));
+  const parameter = matches?.groups.parameter;
+
+  return parameter !== undefined && parameter.length > 0
+    ? parameter
+    : undefined;
+}
+
+/**
+ * @typedef {Object} Link
+ * @property {string} linkText - The link text
+ * @property {string} linkUrl - The link URL
+ */
+
+/**
+ * Extract the link from the meta parameter value
+ *
+ * @example - Returns { linkText: 'More options', linkUrl: 'https://github.com/nodejs/nodejs.org' }
+ * extractLink('[More options](https://github.com/nodejs/nodejs.org)')
+ *
+ * @param {any} link - The value to be extracted
+ *
+ * @return {Link | undefined} - The extracted link object
+ */
+function extractLink(link) {
+  if (typeof link !== 'string') {
+    return;
+  }
+
+  const matches = link.match(/\[(?<text>.*)]\((?<url>.*)\)/);
+  const { text, url } = matches?.groups ?? {};
+
+  if (text === undefined || url === undefined) {
+    return;
+  }
+
+  return {
+    linkText: text,
+    linkUrl: url,
+  };
+}
+
+/** @typedef {import('unist').Node} Node */
+
+/**
+ * Transform code elements using multiple tabs into a `codeTabs` element
+ *
+ * @example - Code tabs supported format
+ * ```mjs|cjs labels="MJS,CJS" link="[More options](https://github.com/nodejs/nodejs.org)"
+ * const { createHmac } = await import('node:crypto');
+ * --------------
+ * const { createHmac } = require('node:crypto');
+ * ```
+ *
+ * @return {function(Node): void}
+ */
+function codeTabsResolver() {
+  return function transformer(tree) {
+    visit(tree, 'element', (node, _, parent) => {
+      const { tagName, children, properties, data } = node;
+
+      if (tagName !== 'code' || children === undefined) {
+        return;
+      }
+
+      const rawCode = children[0]?.value;
+
+      if (SEPARATOR_REGEX.test(rawCode)) {
+        const codeSnippets = rawCode.split(SEPARATOR_REGEX);
+        const className = properties.className.join(' ');
+        const matches = className.match(/language-(?<lang>.*)/);
+
+        const languages = matches?.groups.lang.split('|');
+        const metaLabels = getMetaParameter(data?.meta, 'labels');
+        const metaLink = getMetaParameter(data?.meta, 'link');
+        const link = extractLink(metaLink);
+
+        // Replace the `pre` element with the custom `codeTabs` element
+        // This way, the tabs' structure can be built within MDX components
+        Object.assign(parent, {
+          tagName: 'codeTabs',
+          properties: {
+            labels: metaLabels,
+            languages: languages.join('|'),
+            ...link,
+          },
+          children: languages.map((language, index) => ({
+            type: 'element',
+            // `rehype-pretty-code` only formats code elements as a code block
```

Comment:
But that is correct. We should only transform `pre` elements, no?

Reply:
Yep. The comment is just to explicit the need for the `pre` element before the `code` element. Do you think we can just remove this comment?

Reply:
The comment is quite unclear, imo. Ill wait for you with the new implementation for further feedback.

Reply:
Got it. Well, I believe we can remove the comment to avoid any confusion.